server/src/main/java/org/eclipse/openvsx/search/SearchService.java

```diff
@@ -188,10 +189,24 @@ else if (value > 1.0)
             queryBuilder.withFilter(QueryBuilders.matchPhraseQuery("categories", category));
         }
 
-        // Configure default sorting of results
-        queryBuilder.withSort(SortBuilders.scoreSort());
-        queryBuilder.withSort(SortBuilders.fieldSort("relevance").unmappedType("float").order(SortOrder.DESC));
+        if ("relevance".equals(sortBy)) {
+            queryBuilder.withSort(SortBuilders.scoreSort());
+        }
 
+        if ("relevance".equals(sortBy) || "averageRating".equals(sortBy)) {
+            queryBuilder.withSort(
+                    SortBuilders.fieldSort(sortBy).unmappedType("float").order(SortOrder.fromString(sortOrder)));
+        } else if ("timestamp".equals(sortBy)) {
+            queryBuilder.withSort(
+                    SortBuilders.fieldSort(sortBy).unmappedType("long").order(SortOrder.fromString(sortOrder)));
+        } else if ("downloadCount".equals(sortBy)) {
+            queryBuilder.withSort(
+                    SortBuilders.fieldSort(sortBy).unmappedType("integer").order(SortOrder.fromString(sortOrder)));
+        } else {
+            throw new ErrorResultException(
+                    "sortBy parameter must be 'relevance', 'timestamp', 'averageRating' or 'downloadCount'");
```

Comment:
If I use "asdf" for `sortBy`, I get a status 500 response. The ErrorResultException must be caught in the RegistryAPI method so the endpoint can respond with a proper status and error message.

If I use "asdf" for `sortOrder`, I get a status 500 response with the message "No enum constant org.elasticsearch.search.sort.SortOrder.ASDF" (IllegalArgumentException). This should be replaced with a better message.